node/impl/client/import.go

```diff
@@ -0,0 +1,174 @@
+package client
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"os"
+
+	"github.com/filecoin-project/lotus/build"
+	"github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-cidutil"
+	chunker "github.com/ipfs/go-ipfs-chunker"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
+	files2 "github.com/ipfs/go-ipfs-files"
+	ipld "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipfs/go-unixfs/importer/balanced"
+	ihelper "github.com/ipfs/go-unixfs/importer/helpers"
+	"github.com/ipld/go-car"
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/blockstore"
+	"golang.org/x/xerrors"
+)
+
+// importNormalFileToCARv2 transforms the client's "normal file" to a Unixfs IPLD DAG and writes out the DAG to a CARv2 file at the given output path.
+func (a *API) importNormalFileToCARv2(ctx context.Context, importID uint64, inputFilePath string, outputCARv2Path string) (c cid.Cid, finalErr error) {
+
+	// TODO: We've currently put in a hack to create the Unixfs DAG as a CARv2 without using Badger.
+	// We first transform the Unixfs DAG to a rootless CARv2 file as CARv2 doesen't allow streaming writes without specifying the root upfront and we
+	// don't have the root till the Unixfs DAG is created.
+	//
+	// In the second pass, we create a CARv2 file with the root present using the root node we get in the above step.
+	// This hack should be fixed when CARv2 allows specifying the root AFTER finishing the CARv2 streaming write.
+	tmpCARv2Path, err := a.imgr().NewTempFile(importID)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("failed to create temp CARv2 file: %w", err)
+	}
+	defer os.Remove(tmpCARv2Path) //nolint:errcheck
+
+	tempCARv2Store, err := blockstore.NewReadWrite(tmpCARv2Path, []cid.Cid{})
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("failed to create rootless temp CARv2 rw store: %w", err)
+	}
+	defer tempCARv2Store.Finalize() //nolint:errcheck
+	bsvc := blockservice.New(tempCARv2Store, offline.Exchange(tempCARv2Store))
```

Comment:
We could avoid a lot of write i/o by using an (in-memory) filestore here. This should make imports significantly faster - We'd only do sequential reading here, and likely get most of the file cached in RAM, so in the second pass we'd mostly do sequential writing (yes, SSDs are pretty good at random io too, but most are really bad at mixing reading and writing)

Reply:
@magik6k Can you point me to some code that uses Filestores to do create a Unixfs DAG ? Not sure what you mean here.

Reply:
@magik6k I've addressed this comment by using a filestore backed by an in-memory blockstore in the first pass here. However, I have some concerns around cleaning up this temporary filestore that I've marked as a `TODO` in the code. 
For the cleanup, should we simply iterate over all the keys the filestore has and delete them or is there more to it ?

Reply:
This has been done.